When Rspec test loading fails, adequate error information is not surfaced

### Your environment

- `vscode-ruby-test-adapter` version: 0.6.0
- Ruby version: ruby 2.6.5p114 (2019-10-01 revision 67812) [x86_64-linux]
- VS Code version: 1.39.2
- Operating System: Debian GNU/Linux 10 (buster) (Docker container)
- RSpec version: 3.9.0

### Expected behavior

Errors in encountered by rspec when loading tests (`rspec --dry-run`) should be surfaced in the IDE.

### Actual behavior

Ruby test explorer shows a generic failure message. 
For example, when using a variable defined in a `let(){}` function outside of a describe or context block, I see the following:

A vscode notification, that has some of the output, but not the relevant error message:

![image](https://user-images.githubusercontent.com/1452968/68362901-adaf4380-0174-11ea-8e68-478f45e5b922.png)

Copying that message and pasting it elsewhere to get the full message does not show the error either.

The `Ruby Test Explorer Log` shows the same output as the message:

```
[2019-11-07 03:43:13.674] [INFO] Running dry-run of RSpec test suite with the following command: bundle exec rspec --require /home/baw_web/.vscode-server/extensions/connorshea.vscode-ruby-test-adapter-0.6.0/custom_formatter.rb --format CustomFormatter --order defined --dry-run
[2019-11-07 03:43:13.688] [INFO] Sending autorun event
[2019-11-07 03:44:00.321] [ERROR] Error while finding RSpec test suite: Command failed: bundle exec rspec --require /home/baw_web/.vscode-server/extensions/connorshea.vscode-ruby-test-adapter-0.6.0/custom_formatter.rb --format CustomFormatter --order defined --dry-run
/usr/local/bundle/gems/activesupport-4.2.11.1/lib/active_support/core_ext/object/duplicable.rb:111: warning: BigDecimal.new is deprecated; use BigDecimal() method instead.
The client method is deprecated as of redis-rb 4.0.0, please use the new _clientmethod instead. Support for the old method will be removed in redis-namespace 2.0.
The PGconn, PGresult, and PGError constants are deprecated, and will be
removed as of version 1.0.

You should use PG::Connection, PG::Result, and PG::Error instead, respectively.

Called from /usr/local/bundle/gems/activesupport-4.2.11.1/lib/active_support/dependencies.rb:240:in `load_dependency'
/usr/local/bundle/gems/shoulda-matchers-2.8.0/lib/shoulda/matchers/active_model/validate_inclusion_of_matcher.rb:251: warning: BigDecimal.new is deprecated; use BigDecimal() method instead.
```

There's a lot of rubbish in there but none of it is the error.

If I run the command myself in another console:
```
baw_web@38f11388d3f9:~/baw-server$  bundle exec rspec --require /home/baw_web/.vscode-server/extensions/connorshea.vscode-ruby-test-adapter-0.6.0/custom_formatter.rb --format CustomFormatter --order defined --dry-run

***
Tests must be run in the test envrionment.
The current envrionment `development` has been changed to `test`.
See rails_helper.rb to disable this check
***
/usr/local/bundle/gems/activesupport-4.2.11.1/lib/active_support/core_ext/object/duplicable.rb:111: warning: BigDecimal.new is deprecated; use BigDecimal() method instead.
===> Settings: 'test' loaded from /home/baw_web/baw-server/config/settings/default.yml.
===> BawWorkers::Settings: 'test' loaded from /home/baw_web/baw-server/config/settings/default.yml.
===> Settings: 'test' merged from /home/baw_web/baw-server/config/settings/test.yml.
===> BawWorkers::Settings: 'test' merged from /home/baw_web/baw-server/config/settings/test.yml.
===> Configuration passed validation.
The client method is deprecated as of redis-rb 4.0.0, please use the new _clientmethod instead. Support for the old method will be removed in redis-namespace 2.0.
The PGconn, PGresult, and PGError constants are deprecated, and will be
removed as of version 1.0.

You should use PG::Connection, PG::Result, and PG::Error instead, respectively.

Called from /usr/local/bundle/gems/activesupport-4.2.11.1/lib/active_support/dependencies.rb:240:in `load_dependency'
[Zonebie] Setting timezone: ZONEBIE_TZ="Kyiv"
===> Time zone offset is 7200.
/usr/local/bundle/gems/shoulda-matchers-2.8.0/lib/shoulda/matchers/active_model/validate_inclusion_of_matcher.rb:251: warning: BigDecimal.new is deprecated; use BigDecimal() method instead.
START_OF_TEST_JSON{"version":"3.9.0","messages":["\nAn error occurred while loading ./spec/acceptance/media/media_access_spec.rb.\n\u001B[31mFailure/Error: \u001B[0mprepare_project_logged_in(site)\u001B[0m\n\u001B[31m  `site` is not available on an example group (e.g. a `describe` or `context` block). It is only available from within individual examples (e.g. `it` blocks) or from constructs that run in the scope of an example (e.g. `before`, `let`, etc).\u001B[0m\n\u001B[36m# ./spec/acceptance/media/media_access_spec.rb:19:in `block in \u003Ctop (required)\u003E'\u001B[0m\n\u001B[36m# /usr/local/bundle/gems/rspec_api_documentation-4.8.0/lib/rspec_api_documentation/dsl.rb:26:in `resource'\u001B[0m\n\u001B[36m# ./spec/acceptance/media/media_access_spec.rb:8:in `\u003Ctop (required)\u003E'\u001B[0m\n\u001B[36m# /usr/local/bundle/gems/activesupport-4.2.11.1/lib/active_support/dependencies.rb:268:in `load'\u001B[0m\n\u001B[36m# /usr/local/bundle/gems/activesupport-4.2.11.1/lib/active_support/dependencies.rb:268:in `block in load'\u001B[0m\n\u001B[36m# /usr/local/bundle/gems/activesupport-4.2.11.1/lib/active_support/dependencies.rb:240:in `load_dependency'\u001B[0m\n\u001B[36m# /usr/local/bundle/gems/activesupport-4.2.11.1/lib/active_support/dependencies.rb:268:in `load'\u001B[0m\n"],"examples":[],"summary":{"duration":0.0001013,"example_count":0,"failure_count":0,"pending_count":0,"errors_outside_of_examples_count":1},"summary_line":"0 examples, 0 failures, 1 error occurred outside of examples"}END_OF_TEST_JSON
```

Bingo! In between the *_OF_TEST_JSON markers, the error from rspec exists.

The error reported from rspec is totally user error. This bug is rather that the user error is squashed and not shown anywhere in vscode (as far as I can tell).

I believe the issue is here:

https://github.com/connorshea/vscode-ruby-test-adapter/blob/50885520546feb691d8124f799fc823f604be3fa/src/rspecTests.ts#L41-L49

i think the standard output should probably be parsed before the exception is thrown.

--- a/src/rspecTests.ts
+++ b/src/rspecTests.ts
@@ -40,8 +40,6 @@ export class RspecTests extends Tests {
 
     childProcess.exec(cmd, execArgs, (err, stdout) => {
       if (err) {
-        console.log(err);
-        console.log(stdout);
         this.log.error(`Error while finding RSpec test suite: ${err.message}`);
         // Show an error message.
         vscode.window.showWarningMessage(

--- a/src/rspecTests.ts
+++ b/src/rspecTests.ts
@@ -40,10 +40,31 @@ export class RspecTests extends Tests {
 
     childProcess.exec(cmd, execArgs, (err, stdout) => {
       if (err) {
+        console.log(err);
+        console.log(stdout);
         this.log.error(`Error while finding RSpec test suite: ${err.message}`);
         // Show an error message.
-        vscode.window.showWarningMessage("Ruby Test Explorer failed to find an RSpec test suite. Make sure RSpec is installed and your configured RSpec command is correct.");
-        vscode.window.showErrorMessage(err.message);
+        vscode.window.showWarningMessage(
+          "Ruby Test Explorer failed to find an RSpec test suite. Make sure RSpec is installed and your configured RSpec command is correct.",
+          "View error message"
+        ).then(selection => {
+          if (selection === "View error message") {
+            let outputJson = JSON.parse(Tests.getJsonFromOutput(stdout));
+            let outputChannel = vscode.window.createOutputChannel('Ruby Test Explorer Error Message');
+
+            if (outputJson.messages.length > 0) {
+              let outputJsonString = outputJson.messages.join("\n\n");
+              let outputJsonArray = outputJsonString.split("\n");
+              outputJsonArray.forEach((line: string) => {
+                outputChannel.appendLine(line);
+              })
+            } else {
+              outputChannel.append(err.message);
+            }
+            outputChannel.show(false);
+          }
+        });
+
         throw err;
       }
       resolve(stdout);

--- a/src/tests.ts
+++ b/src/tests.ts
@@ -44,7 +44,7 @@ export abstract class Tests {
     let output = await this.initTests();
     this.log.debug('Passing raw output from dry-run into getJsonFromOutput.');
     this.log.debug(`${output}`);
-    output = this.getJsonFromOutput(output);
+    output = Tests.getJsonFromOutput(output);
     this.log.debug('Parsing the below JSON:');
     this.log.debug(`${output}`);
     let testMetadata;
@@ -116,7 +116,7 @@ export abstract class Tests {
    * @param output The output returned by running a command.
    * @return A string representation of the JSON found in the output.
    */
-  protected getJsonFromOutput(output: string): string {
+  static getJsonFromOutput(output: string): string {
     output = output.substring(output.indexOf('START_OF_TEST_JSON{'), output.lastIndexOf('}END_OF_TEST_JSON') + 1);
     // Get rid of the `START_OF_TEST_JSON` and `END_OF_TEST_JSON` to verify that the JSON is valid.
     return output.substring(output.indexOf("{"), output.lastIndexOf("}") + 1);
@@ -422,7 +422,7 @@ export abstract class Tests {
       this.testStatesEmitter.fire(<TestEvent>{ type: 'test', test: node.id, state: 'running' });
 
       let testOutput = await this.runFullTestSuite();
-      testOutput = this.getJsonFromOutput(testOutput);
+      testOutput = Tests.getJsonFromOutput(testOutput);
       this.log.debug('Parsing the below JSON:');
       this.log.debug(`${testOutput}`);
       let testMetadata = JSON.parse(testOutput);
@@ -441,7 +441,7 @@ export abstract class Tests {
 
       let testOutput = await this.runTestFile(`${node.file}`);
 
-      testOutput = this.getJsonFromOutput(testOutput);
+      testOutput = Tests.getJsonFromOutput(testOutput);
       this.log.debug('Parsing the below JSON:');
       this.log.debug(`${testOutput}`);
       let testMetadata = JSON.parse(testOutput);
@@ -473,7 +473,7 @@ export abstract class Tests {
         // VS Code and 1-indexed for RSpec/Minitest.
         let testOutput = await this.runSingleTest(`${node.file}:${node.line + 1}`);
 
-        testOutput = this.getJsonFromOutput(testOutput);
+        testOutput = Tests.getJsonFromOutput(testOutput);
         this.log.debug('Parsing the below JSON:');
         this.log.debug(`${testOutput}`);
         let testMetadata = JSON.parse(testOutput);

